### ISSUE TITLE
Add auth field to the VisitOptions TS definition

### ISSUE THREAD

**Issue by daniel-grotting** [cypress-io/cypress#1897]
### Current behavior:

The current type definition in VisitOptions is missing the auth field. If you try to use in a TypeScript project it will fail type validation.

### Desired behavior:

To be able to use the auth option in cy.visit()

### Steps to reproduce:

In a TypeScript project try to add `auth` to the options object in cy.visit(), like 
```js
cy.visit("/", { auth: {
	user: john,
	pass: password,
  }
})
```
will fail type validation.

### Versions

TypeScript

Suggested solution: 

Add:
```js
interface VisitOptions extends Loggable, Timeoutable {
    /**
     * Adds Basic Authorization headers
     * 
     * @param {ObjectLike} auth
     */
    auth: ObjectLike
```
in `cli/types.d.ts`

I was about to submit a PR but I have to wait to get the CLA signed

**Comment by jennifer-shehane** [comment]
We have our pull requests configured to automatically ask you for CLA signature. But, you should be able to open one without signing it right away if you needed to review the CLA.

**Comment by brian-mann** [comment]
Released in [`3.0.2`](https://docs.cypress.io/guides/references/changelog.html#3-0-2).
